src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -74,29 +74,34 @@ def _cache_dir(self):
 
   def select_interpreter_for_targets(self, targets):
     """Pick an interpreter compatible with all the specified targets."""
-    tgts_with_compatibilities = []
+    tgts_by_compatibilities = {}
     filters = set()
     for target in targets:
-      if isinstance(target, PythonTarget) and target.compatibility:
-        tgts_with_compatibilities.append(target)
-        filters.update(target.compatibility)
+      if isinstance(target, PythonTarget):
+        c = self._python_setup.compatibility_or_constraintstar(target)
```

Comment:
typo: _python_setup.compatibility_or_constraintstar

Reply:
Blargh, thanks. Extracted the helper after getting things working.